keras_nlp/layers/masked_lm_mask_generator.py

```diff
@@ -81,16 +81,76 @@ class MaskedLMMaskGenerator(keras.layers.Layer):
     Examples:
 
     Basic usage.
-    >>> masker = keras_nlp.layers.MaskedLMMaskGenerator(
-    ...     vocabulary_size=10, mask_selection_rate=0.2, mask_token_id=0,
-    ...     mask_selection_length=5)
-    >>> masker(tf.constant([1, 2, 3, 4, 5]))
-
-    Ragged Input:
-    >>> masker = keras_nlp.layers.MaskedLMMaskGenerator(
-    ...     vocabulary_size=10, mask_selection_rate=0.5, mask_token_id=0,
-    ...     mask_selection_length=5)
-    >>> masker(tf.ragged.constant([[1, 2], [1, 2, 3, 4]]))
+    ```python
+    masker = keras_nlp.layers.MaskedLMMaskGenerator(
+        vocabulary_size=10,
+        mask_selection_rate=0.2,
+        mask_token_id=0,
+        mask_selection_length=5
+    )
+    # Dense input.
+    masker(tf.constant([1, 2, 3, 4, 5]))
+
+    # Ragged input.
+    masker(tf.ragged.constant([[1, 2], [1, 2, 3, 4]]))
+    ```
+
+    An end-to-end masked language model training using masked language mask
+    generator.
+    ```python
+    train_data = tf.constant([
```

Comment:
Some high level comments...

I think showing a more fleshed out usage is OK to do here, but I would scale it down significantly from what we are showing here.

We do not want to show `keras_nlp.models` usage here in our lower level layers. And we probably do not want to show `fit()` either. We should generally try to keep our usages fairly local to the symbols in question, so we avoid a ton of cross cutting dependencies when we update a symbol usage.

I might instead just show something that looks closer to a real world example, but is still just focused on the layer itself.

```python
pad_id, cls_id, sep_id, mask_id = 0, 1, 2, 3
batch = [
    [cls_id,     4,     5,      6, sep_id,    7,    8, sep_id, pad_id, pad_id],
    [cls_id,     4,     5, sep_id,      6,    7,    8,      9, sep_id, pad_id],
]
# the rest of the MaskedLMMaskGenerator and invocation
```

Reply:
like last push?

Reply:
Ah yes perhaps, sorry about that!

This may be me needing to sync up with @chenmoneygithub more. I do not think we want the examples here to depend on a totally separate part of our API, I will talk with @chenmoneygithub about it.

Reply:
Just talked with @chenmoneygithub, apologies for all the confusion here. We are scaling up our contributors, so our wires may get crossed a few times :)

But this latest version looks good to me! Thanks very much!

Reply:
No problem. Writing docstrings and changes help me to get familiar with the library more. Thanks for the feedback.
One thing: should I run that /gcbrun or a mentor should run it ?

Reply:
@abuelnasr0 good question. We are playing with different options here, but `/gcbrun` would need to be added to project by a maintainer. Unclear if we are going to require it long term, but no action should be needed from you!